lib/tiled/map.rb

```diff
@@ -89,6 +89,28 @@ def pixelheight
       return height * tileheight
     end
 
+    #gets the first object layer with the passed name of `layer_name`
+    #then looks in that layer for the first object that has the name `object_name`
+    #then returns that object
+    def object_by_name(layer_name, object_name)
+      
+      named_layer = @object_groups.reject do | layer |
+        layer.name != layer_name
+      end.first
+
+      raise ObjectLayerNotFound, "The object layer \"#{layer_name}\" was not found in map \"#{@path}\"" if named_layer.nil?
```

Comment:
Right now all methods return `nil` if desired object/tile is not found, it is better to save this approach.

But I think it is good to have a method that raises errors in case an object is not found. I love the approach when the regular method is returning `nil` and the bang method raises errors:
```ruby
def object_by_name(object_name)
  # return nil and not raise errors
end

def object_by_name!(object_name)
  raise SomeError if the_object.nil?
end
```

Do you think it makes sense?

Reply:
I think with the release of v0.4.0 this can be closed for now.